compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1455,10 +1443,28 @@ object desugar {
 
           val applyVParams = vargs.asInstanceOf[List[ValDef]]
             .map(varg => varg.withAddedFlags(mods.flags | Param))
-          New(Template(emptyConstructor, List(polyFunctionTpt), Nil, EmptyValDef,
-            List(DefDef(nme.apply, applyTParams, List(applyVParams), TypeTree(), res))
-          ))
+            New(Template(emptyConstructor, List(polyFunctionTpt), Nil, EmptyValDef,
+              List(DefDef(nme.apply, applyTParams, List(applyVParams), TypeTree(), res))
+              ))
         }
+      case _ =>
+        EmptyTree //may happen for erroneous input
```

Comment:
If the input is erroneous, shouldn't we report an error here ? (also leave a space between `//` and the first character in a comment)

Reply:
An error will already have been reported, this is to prevent `PolyFunction` trees from being passed along and blowing up downstream. I've added a comment to that effect (and fixed the whitespace).